src/mame/efo/cedar_magnet.cpp

```diff
@@ -265,6 +265,440 @@ MASTER BOARD
+Does 0xFFFF do something more than just act as ram? 
+
+088E: 00        	nop
+088F: F3        	di
+0890: 3E 54     	ld   a,$54
+0892: F6 28     	or   $28
+0894: 4F        	ld   c,a
+0895: 21 FF FF  	ld   hl,$FFFF
+0898: 46        	ld   b,(hl)
+0899: 70        	ld   (hl),b
+089A: 70        	ld   (hl),b
+089B: 70        	ld   (hl),b
+089C: ED 50     	in   d,(c)   << port 7c
+089E: 70        	ld   (hl),b
+089F: ED 58     	in   e,(c)  << port 7c
+08A1: 7B        	ld   a,e
+08A2: E6 7F     	and  $7F
+08A4: 20 08     	jr   nz,$08AE
+08A6: 2E 1A     	ld   l,$1A
+08A8: 26 FF     	ld   h,$FF
+08AA: 46        	ld   b,(hl)
+08AB: 70        	ld   (hl),b
+08AC: 18 E7     	jr   $0895
+
+08AE: 7A        	ld   a,d
+08AF: 07        	rlca
+08B0: 07        	rlca
+08B1: 07        	rlca
+08B2: 07        	rlca
+08B3: AA        	xor  d
+08B4: 07        	rlca
+08B5: 07        	rlca
+08B6: 7A        	ld   a,d
+08B7: 17        	rla
+08B8: AB        	xor  e
+08B9: E6 7F     	and  $7F
+08BB: C8        	ret  z
+08BC: 3E FF     	ld   a,$FF
+08BE: 32 8C 08  	ld   ($088C),a
+08C1: C9        	ret
+
+
+0xFF - out, sound latch? Coin counter? Written after reading coins at least.
+
+
+
+Z80 sound @ 3.9986 Mhz
+----------------------
+
+Memory map
+0000-FFFF Ram (full 64kb or mirrored? check mem chip count on pcb)
+
+J2 sound board connector (labeled as J3 on main board)
+
+1 EDGE25 PARTS SPEAKER +
+2 EDGE25 SOLDER SPEAKER -
+3 GND
+4 GND
+5 +5v
+6 +12v
+7 Ic28 74244 1 19 /ENA   ic29  74244 1 19 /ENA   ic32 74241 1 19 ENA   ic33 7432 3 12
+8 ic34 74HC14  9 -> IC35 74HC00 5 -> IC19 Z80 CPU 25 /Busreq
+9 ic34 74HC14 11 -> IC22 IC23 Z80CTC 17 /Reset
+
+
+Z80 sound port map
+
+0x00 - 0x03 AM_DEVREADWRITE("ctc0", z80ctc_device, read, write)
+0x04 -0x07 AM_DEVREADWRITE("ctc1", z80ctc_device, read, write)
+
+0x0c - 0x0d AM AY8910 #1
+0x10 - 0x11 AM_AY8910 #2
+    
+0x14 input from main cpu? (maybe read what was written to 0xff?)
+
+There’s also an M5205 ADPCM playback chip, not seen writes for it.
+
+
+Sound board CPU iorq's
+
+Z80 IORQ goes into IC11 (74138) /ENA1 control together with ENA3 (the output of IC35 (7400) watching over the z80 RD/WR signals), /ENA2 is gnd.
+Z80 A0 -> 28 IC20/21 AY-3-8910A BC2 (BC1 is grounded)
+Z80 A0 A1 -> 18/19 IC22/23 Z80 CTC CS0 CS1 Selects
+Z80 A2 A3 A4 -> IC11 decoder  inputs A0 A1 A2
+
+Decoder IC11 outputs as follows:
+
+15 /Y0 -> 16 IC22 Z80CTC /CE
+14 /Y1 -> 16 IC23 Z80CTC /CE
+13 /Y2 -> 3 IC8 (7404 inverter) -> 3 IC13 HCF40105BE (FIFO) SI (Shift in)
+12 /Y3 -> 1 IC8 (7404 inverter) -> 27 IC20 AY-3-8910A BDIR
+11 /Y4 -> 5 IC8 (7404 inverter) -> 27 IC21 AY-3-8910A BDIR
+10 /Y5 -> 13 IC12 (7474 FF) CLEAR & 1 IC31 (74HC374 FF) /OE
+9 /Y6 -> N.C.
+7 /Y7 -> N.C.
+
+
+
+Z80 sprite notes
+----------------
+
+Port maps
+
+  0xc0 - 0xc3 AM_DEVREADWRITE("z80pio0", z80pio_device, read_alt, write_alt)
+  0xc4 - 0xc7 AM_DEVREADWRITE("z80pio1", z80pio_device, read_alt, write_alt)
+  0xc8 - 0xcb AM_DEVREADWRITE("z80pio2", z80pio_device, read_alt, write_alt)
```

Comment:
Legacy address map macros?  Seriously?  Please describe what this stuff is rather than using code that won’t even compile if you try to use it.